concepts/match-basics/about.md

```diff
@@ -56,66 +64,85 @@ fn add_one(x: Option<u8>) -> Option<u8> {
 
 ```rust
 fn add_one(x: Option<u8>) -> Option<u8> {
     match x {
         Option::Some(val) => Option::Some(val + 1),
     }
+     // this code does not compile, as we forgot to handle the `None` case
 }
 ```
 
-This kind of bug is easier for the compiler to spot because in this case, the possibility that the `x` value could be a null value was not handled. The revert error looks like this error:
-
-```sh
-$ scarb cairo-run
-   Compiling missing_match_arm v0.1.0 (home/Scarb.toml)
-error: Missing match arm: `None` not covered.
- --> home/src/lib.cairo:5:5
-    match x {
-    ^*******^
-
-error: could not compile `missing_match_arm` due to previous error
-error: `scarb metadata` exited with error
-
-```
-
 The error shows that Cairo is aware of other uncovered possible cases that x could be. This error spells out the possibility of the `x` value being null and protect us from assuming that `x` will never be null.
 
-## Catch-all pattern with `_` wildcard
+## Catch-all Pattern with `_` Wildcard
 
-Understanding how the `_` wildcard works is essential in order to effectively handle all cases. This is most times added as the last arm of a `match` expression. This is a special pattern whose arm of code is executed only after the critical patterns.
+Using enums, we can also take special actions for a few particular values, but for all other values perform one default action by adding a new arm with `_` as the pattern for the last arm of the match expression. The `_` pattern is a special pattern that matches any value and does not bind to that value.
 
-The allow_red function receives a parameter and return true if red and false for any other color instance.
+The `is_red` function receives a parameter and return `true` if the color is `Red` and `false` for any other color:
 
 ```rust
-fn allow_red(color: Color) -> bool {
+fn is_red(color: Color) -> bool {
     match color {
         Color::Red => true,
         _ => false,
     }
 }
 ```
 
-## Match patterning with the `or` (`|`) Operator
+## Pattern Matching with the `|` Operator
 
-The Cairo `match` control flow construct allows for matching a value against multiple patterns handled by the `or` operator (`|`). With the use of the `or` operator in a match patterning, it helps achieve code simplicity.
+The Cairo `match` control flow construct allows for matching a value against multiple patterns using the `|` (_or_) operator. This helps simplify the code.
 
-For example, using the `Color` enum above, we want to write a function that returns true for `Red` or `Green` colors and false for other colors.
+For example, using the `Color` enum, we want to write a function that returns `true` for `Red` or `Green` colors and `false` for other colors:
 
 ```rust
-fn allow_red_and_green(color: Color) -> bool {
+fn is_red_or_green(color: Color) -> bool {
     match color {
         Color::Red | Color::Green => true,
         _ => false,
     }
 }
 ```
+
```

Comment:
Write a ["Matching Tuples"](https://book.cairo-lang.org/ch06-02-the-match-control-flow-construct.html#matching-tuples) section, shorten it as much as you can.